app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt

```diff
@@ -0,0 +1,153 @@
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.github.se.eduverse.model.Widget
+import com.github.se.eduverse.viewmodel.DashboardViewModel
+
+@SuppressLint("UnusedMaterialScaffoldPaddingParameter")
+@Composable
+fun DashboardScreen(viewModel: DashboardViewModel, userId: String) {
+  val widgetList by viewModel.widgetList.collectAsState()
+  val availableWidgets by viewModel.availableWidgets.collectAsState()
+
+  LaunchedEffect(userId) { viewModel.fetchWidgets(userId) }
+
+  var showDropdown by remember { mutableStateOf(false) }
+  var deleteWidgetId by remember {
+    mutableStateOf<String?>(null)
+  } // State to track which widget shows delete icon
+
+  Scaffold(
+      floatingActionButton = {
+        FloatingActionButton(
+            onClick = {
+              showDropdown = !showDropdown
+              if (showDropdown) {
+                viewModel.fetchAvailableWidgets()
+              }
+            },
+            modifier = Modifier.testTag("add_widget_button")) {
+              Icon(Icons.Default.Add, contentDescription = "Add Widget")
+            }
+      },
+      bottomBar = { BottomNavigationPlaceholder() },
+      modifier = Modifier.testTag("dashboard_screen")) {
+        Column {
+          LazyColumn(modifier = Modifier.testTag("widget_list")) {
+            items(widgetList) { widget ->
+              WidgetCard(
+                  widget = widget,
+                  showDeleteIcon =
+                      deleteWidgetId ==
+                          widget.widgetId, // Only show delete icon for the long-pressed widget
+                  onLongPress = {
+                    deleteWidgetId =
+                        if (deleteWidgetId == widget.widgetId) null else widget.widgetId
+                  },
+                  onDeleteClick = {
+                    viewModel.removeWidget(userId, widget.widgetId)
+                    Log.d("DashboardScreen", "Widget removed: ${widget.widgetId}")
+                    viewModel.fetchWidgets(userId) // Fetch updated list after removing widget
+                    deleteWidgetId = null // Reset delete icon state after deletion
+                  })
+            }
+          }
+
+          if (showDropdown) {
+            LazyColumn(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
+              items(availableWidgets) { widget ->
+                AvailableWidgetCard(
+                    widget = widget,
+                    onClick = {
+                      viewModel.addWidget(userId, widget)
+                      showDropdown = false
+                    })
```

Comment:
I feel like it would make more sense to use a DropdownMenu instead of a LazyColumn. Additionaly, I think if you have more widgets than space on the screen, the second column (with which kind of widget we want to add) will not appear on the screen

Reply:
yes you are right ! working on it

Reply:
done